core/consensus/src/wal.rs

```diff
@@ -129,6 +154,177 @@ impl SignedTxsWAL {
     }
 }
 
+#[test]
+fn a() {}
+
+#[derive(Debug)]
+pub struct ConsensusWal {
+    path: PathBuf,
+}
+
+impl ConsensusWal {
+    pub fn new<P: AsRef<Path>>(path: P) -> Self {
+        if !path.as_ref().exists() {
+            fs::create_dir_all(&path).expect("Failed to create wal directory");
+        }
+
+        ConsensusWal {
+            path: path.as_ref().to_path_buf(),
+        }
+    }
+
+    #[muta_apm::derive::tracing_span(kind = "consensus_wal")]
+    pub fn update_overlord_wal(&self, ctx: Context, info: Bytes) -> ProtocolResult<()> {
```

Comment:
You should create an overlord wal file per height, and append the later info to the file. This can make the code brief.

Reply:
No, that's useless because on each height, there could be more than one write operate on wal.
And modification of add height param of interface has to be done